docs/website/docs/general-usage/schema-evolution.md

```diff
@@ -0,0 +1,205 @@
+
+## How to control evolution
+
+`dlt` allows schema evolution control via its schema and data contracts. Refer to our **[documentation](https://dlthub.com/docs/general-usage/schema-contracts)** for details.
+
+### How to test for removed columns - applying “not null” constraint
+
+A column not existing, and a column being null, are two different things. However, when it comes to APIs and json, it’s usually all treated the same - the key-value pair will simply not exist. 
+
+To remove a column, exclude it from the output of the resource function. Subsequent data inserts will treat this column as null. Verify column removal by applying a not null constraint. For instance, after removing the "room" column, apply a not null constraint to confirm its exclusion.
+
+```python
+import dlt
+
+# Define a data resource using dlt.resource with a schema contract
+@dlt.resource(schema_contract={"tables": "evolve", "columns": "evolve"}, columns={"room": {"data_type": "integer", "nullable": "false"}})
+def data_resource():
+    yield {
+    "organization": "Tech Innovations Inc.",
+    "address": {
+        'building': 'r&d' 
+        #"room": 7890,
+    },
+    "Inventory": [
+        {"name": "Plasma ray", "inventory nr": 2411},
+        {"name": "Self-aware Roomba", "inventory nr": 268},
+        {"name": "Type-inferrer", "inventory nr": 3621}
+    ]
+}
+```
+During pipeline execution a data validation error indicates that a removed column is being passed as null.
+
+## Some schema changes in the data
+
+The data in the pipeline mentioned above is modified.
+
+```python
+{
+    "organization": "Tech Innovations Inc.",
+    "CEO": "Alice Smith",
+    "address": {'main_block': 'r&d'},
+    "Inventory": [
+        {"name": "Plasma ray", "inventory nr": "AR2411"}, 
+        {"name": "Self-aware Roomba", "inventory nr": "AR268"}, 
+        {
+            "name": "Type-inferrer", "inventory nr": "AR3621",
+            "details": {
+                "category": "Computing Devices", 
+                "id": 369,
+                "specifications": [{ 
+                    "processor": "Quantum Core",
+                    "memory": "512PB"
+                }]
+            }
+        }
+    ]
+}
+```
+The schema of the data above is loaded to the destination as follows:
+<iframe width="560" height="315" src='https://dbdiagram.io/e/65e80b31cd45b569fba33169/65e81055cd45b569fba3aa20'> </iframe>
+
+## What did the schema evolution engine do?
+
+The schema evolution engine in the `dlt` library is designed to handle changes in the structure of your data over time. For example: 
+
+- As above in continuation of the inferred schema, the “specifications” are nested in "details” which are nested in “Inventory” all under table name “org”. So the table created for projects is "org__inventory__details__specifications".
```

Comment:
again the mess with column names, all column names in the article should be formatted as code, because it's not a plain text
![image](https://github.com/dlt-hub/dlt/assets/20367975/c9e2f489-dfa1-48f8-94df-85fc23e448c4)

Reply:
done.